test/xbgpu/test_engine.py

```diff
@@ -1452,3 +1573,150 @@ def get_stream_status(stream_name: str) -> bool:
 
             await client.request("capture-start", output.name)
             assert get_stream_status(output.name) is True, f"Stream {output.name} is still disabled"
+
+    @DEFAULT_PARAMETERS
+    @pytest.mark.parametrize("n_x_streams_to_stop, n_b_streams_to_stop", [(2, 0), (1, 1), (0, 2)])
+    async def test_capture_stop_some_streams(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        mock_recv_streams: list[spead2.InprocQueue],
+        mock_send_stream: list[spead2.InprocQueue],
+        client: aiokatcp.Client,
+        n_ants: int,
+        n_channels_per_substream: int,
+        frequency: int,
+        n_samples_between_spectra: int,
+        n_spectra_per_heap: int,
+        heap_accumulation_threshold: list[int],
+        corrprod_outputs: list[XOutput],
+        beam_outputs: list[BOutput],
+        n_x_streams_to_stop: int,
+        n_b_streams_to_stop: int,
+    ) -> None:
+        """Test capture-stop request on only B-engine data streams.
+
+        Issue a `?capture-stop` request at some point into data processing and
+        check the corresponding streams only have partial data transmission.
+        Also ensure that data is completely received for data streams that did
+        not receive a `?capture-stop` request.
+
+        This test is carried out in combinations to fully exercise stop logic in
+        both X- and BPipelines.
+        """
+        assert n_x_streams_to_stop <= len(corrprod_outputs)
+        assert n_b_streams_to_stop <= len(beam_outputs)
+        n_batches = heap_accumulation_threshold[0]
+        # NOTE This value is arbitrarily chosen, but must be less than
+        # `n_batches` / `HEAPS_PER_FENGINE_PER_CHUNK` to actually take effect.
+        # A stop index <= 0 (zero) is treated as invalid/an indication to proceed
+        # as normal for the following reasons:
+        # - Using index zero (0) to obtain a subset of results yields an empty array.
```

Comment:
It's probably not a particularly important case to test, but why is that an issue? numpy supports empty arrays.